### PR TITLE
proper xml is now required for saving exam pipeline tests

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/ingest-pipeline/ingest-pipeline.support.ts [hunk 1]
```diff
@@ -44,3 +47,37 @@ export function getOrAppendAsyncScript(
...
+ * Identifies and returns all the XML syntax errors for the provided XML string value
+ *
+ * @param value The xml string to check for errors
+ */
+export function getXmlParserErrors(value: string): string[] {
+  const normalizedValue = (value || '').trim();
+  const dom = domParser.parseFromString(normalizedValue, 'text/xml');
+  return Array.from(
+    dom.documentElement.getElementsByTagName('parsererror')
+  ).map(element => element.textContent);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/ingest-pipeline/ingest-pipeline.support.ts]: didn't realize there was such a browser native way of doing xml validation. very convenient

File: webapp/src/main/webapp/src/app/admin/ingest-pipeline/model/pipelines.ts [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
-function isValidPipelineTextText(value: string): boolean {
-  return !isBlank(value) && value.length <= maximumTextLength;
+function getInputTypeValidators(inputType: string): ValidatorFn[] {
+  if (inputType === 'xml') {
+    return [xml];
+  }
+  return [];
 }
 
-export function isValidPipelineTest(value: PipelineTest): boolean {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/ingest-pipeline/model/pipelines.ts]: now this method literally makes a form group to perform validation.
if i did the component setup again i would be passing around wrappers with a handle on a form group so i wouldn't need to create these on the fly.
At least now the validation is using the same code as the form itself so there will be no chance of inconsistency

File: webapp/src/main/webapp/src/app/admin/ingest-pipeline/model/pipelines.ts [hunk 3]
```diff
@@ -1,18 +1,47 @@
...
+}
+
+export function pipelineTestFormGroup(inputType: string): FormGroup {
+  const inputOutputValidators: ValidatorFn[] = getInputOutputValidators(
+    inputType
   );
+  return new FormGroup({
+    name: new FormControl(''),
+    input: new FormControl('', inputOutputValidators),
+    output: new FormControl('', inputOutputValidators)
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/admin/ingest-pipeline/model/pipelines.ts]: Reactive forms eh?

Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/ingest-pipeline/model/pipelines.ts]: yeeah boii ; p

**Pull request merged**